modules/renter/hostdb/hostdb.go

```diff
@@ -115,3 +119,22 @@ func New(cs modules.ConsensusSet, wallet modules.Wallet, tpool modules.Transacti
 
 	return hdb, nil
 }
+
+// newHostDB creates a HostDB using the provided dependencies. Unlike New, it
+// does not have any side effects (i.e. it does not spawn background threads,
+// perform I/O, or call stateful methods of its dependencies.)
+func newHostDB(w hdbWallet, tp hdbTransactionPool, d hdbDialer, s hdbSleeper, p hdbPersister, l hdbLogger) *HostDB {
+	return &HostDB{
+		wallet:  w,
+		tpool:   tp,
+		dialer:  d,
+		sleeper: s,
+		persist: p,
+		log:     l,
+
+		contracts:   make(map[types.FileContractID]hostContract),
+		activeHosts: make(map[modules.NetAddress]*hostNode),
+		allHosts:    make(map[modules.NetAddress]*hostEntry),
+		scanPool:    make(chan *hostEntry, scanPoolSize),
+	}
+}
```

Comment:
I definitely think that if you are going to be mocking, you need to be actually hitting the mocks. If you are making a hostdb during testing it should be calling MkdirAll, but that could be a mocked function itself.

But you want to make sure that the logic is as close as possible to the way the hostdb normally runs. You remove the dependencies so you can test the hostdb in isolation, but you don't want to deconstruct how the hostdb itself functions.